### ISSUE TITLE
Exception handling with corrupted videos

### ISSUE THREAD

**Issue by sbebo** [dmlc/decord#25]
Hi,

I'm finding some issues using decord when a file content is corrupted.
This is the stack trace:

```
[h264 @ 0x558362b3b600] Invalid NAL unit 0, skipping.
[h264 @ 0x558362b3b600] cabac decode of qscale diff failed at 32 20
[h264 @ 0x558362b3b600] error while decoding MB 32 20, bytestream 2173
[h264 @ 0x558362b3b600] Frame num change from 7 to 11
[h264 @ 0x558362b3b600] decode_slice_header error
[h264 @ 0x558362b3b600] concealing 2017 DC, 2017 AC, 2017 MV errors in P frame
[h264 @ 0x558362b3b600] Invalid NAL unit size (1731009324 > 12203).
[h264 @ 0x558362b3b600] Error splitting the input into NAL units.
terminate called after throwing an instance of 'dmlc::Error'
  what():  [15:03:42] /io/decord/src/video/ffmpeg/threaded_decoder.cc:178: Check failed: avcodec_send_packet(dec_ctx_.get(), pkt.get()) >= 0 (-1094995529 vs. 0) Thread worker: Error sending packet.

Stack trace returned 10 entries:
[bt] (0) /home/leo/miniconda3/envs/decord/lib/python3.7/site-packages/decord/libdecord.so(dmlc::StackTrace(unsigned long)+0x7f) [0x7fb6f6ad4ef7]
[bt] (1) /home/leo/miniconda3/envs/decord/lib/python3.7/site-packages/decord/libdecord.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x20) [0x7fb6f6ad51b2]
[bt] (2) /home/leo/miniconda3/envs/decord/lib/python3.7/site-packages/decord/libdecord.so(decord::ffmpeg::FFMPEGThreadedDecoder::WorkerThread()+0x497) [0x7fb6f6b358ab]
[bt] (3) /home/leo/miniconda3/envs/decord/lib/python3.7/site-packages/decord/libdecord.so(void std::_Mem_fn<void (decord::ffmpeg::FFMPEGThreadedDecoder::*)()>::operator()<, void>(decord::ffmpeg::FFMPEGThreadedDecoder*) const+0x65) [0x7fb6f6b3dd51]
[bt] (4) /home/leo/miniconda3/envs/decord/lib/python3.7/site-packages/decord/libdecord.so(void std::_Bind_simple<std::_Mem_fn<void (decord::ffmpeg::FFMPEGThreadedDecoder::*)()> (decord::ffmpeg::FFMPEGThreadedDecoder*)>::_M_invoke<0ul>(std::_Index_tuple<0ul>)+0x43) [0x7fb6f6b3dc83]
[bt] (5) /home/leo/miniconda3/envs/decord/lib/python3.7/site-packages/decord/libdecord.so(std::_Bind_simple<std::_Mem_fn<void (decord::ffmpeg::FFMPEGThreadedDecoder::*)()> (decord::ffmpeg::FFMPEGThreadedDecoder*)>::operator()()+0x1b) [0x7fb6f6b3db5d]
[bt] (6) /home/leo/miniconda3/envs/decord/lib/python3.7/site-packages/decord/libdecord.so(std::thread::_Impl<std::_Bind_simple<std::_Mem_fn<void (decord::ffmpeg::FFMPEGThreadedDecoder::*)()> (decord::ffmpeg::FFMPEGThreadedDecoder*)> >::_M_run()+0x1c) [0x7fb6f6b3da9e]
[bt] (7) /home/leo/miniconda3/envs/decord/lib/python3.7/site-packages/decord/libdecord.so(+0x16cce0) [0x7fb6f6b3ece0]
[bt] (8) /lib/x86_64-linux-gnu/libpthread.so.0(+0x76db) [0x7fb705dfa6db]
[bt] (9) /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7fb705b2388f]


[1]    21258 abort (core dumped)  python test.py
```
I'm running this simple script:

```python
from decord import VideoReader
vr = VideoReader("corruptedvideo.mp4")
vr.get_batch(list(range(70)))
```
and the video is corrupted around second 2 (@ 30fps).
The failure of course is expected, but the process is killed with an abort.
opencv is able to handle the issue, failing (more or less) gracefully.

**Comment by zhreshold** [comment]
closed by #33

**Comment by sbebo** [comment]
Awesome! Now waiting for a release, then :-)
